core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/AnalysisService.java

```diff
@@ -119,9 +120,12 @@ public List<String> getAnalysisDatasourceIds() {
     Set<String> ids = metadataDomainRepository.getDomainIds();
     for ( MondrianCatalog mondrianCatalog : mockMondrianCatalogList ) {
       String domainId = mondrianCatalog.getName() + METADATA_EXT;
-      if ( !ids.contains( domainId ) || !isDSWDatasource( domainId ) ) {
+      String createdWithProperty = mondrianCatalog.getConnectProperties().get( "CreatedWith" );
```

Comment:
I believe It would be best to make a more generalized property instead of the logic being coupled to SME. This makes it easier for others to use it if they have similar needs.

Here we are filtering Catalogs that shouldn't be listed / visible / accessible through data access, or if we invert the logic, that should be ignored by data access. So perhaps using a property name `DataAccessNotListedCatalog=true` if we're just considering the listing, or `DataAccessIgnoreCatalog=true`if we would consider all functionality.  From the code it seems that **only** the listing functionality is filtering the catalogs, which I confess I find odd. So even if a catalog isn't returned in a list request you can still delete it.

Reply:
Updated